pkg/dependencies/factory.go

```diff
@@ -48,7 +48,7 @@ type Dependencies struct {
 	DockerClient              *executables.Docker
 	Kubectl                   *executables.Kubectl
 	Govc                      *executables.Govc
-	Cmk                       *executables.Cmk
+	Cmks                      map[string]cloudstack.ProviderCmkClient
```

Comment:
why do we need to arrange this as a map instead of a list? What additional value do we get from the labeling?

Reply:
Never mind, I think a map is the right move so that we can match these Profiles to an Availability Zone for preflight checking

Reply:
why are we using ProviderCmkClient instead of the executables interface?

Reply:
ProviderCmkClient and executables.Cmk are compliant, however, it seems that the maps of them are not compliant in Golang. When I changed Cmks' type to map[string]*executables.Cmk I got this compile error. 

```
pkg/dependencies/factory.go:270:19: cannot use f.dependencies.Cmks (type map[string]*executables.Cmk) as type map[string]cloudstack.ProviderCmkClient in argument to cloudstack.NewProvider
```

Reply:
I guess this is an OOP issue: https://stackoverflow.com/questions/46433860/golang-using-a-defined-interface-in-a-map-value. Maybe it makes sense to contain the map entirely to be inside the Cmk struct as a map of executable objects

Reply:
My concern with using ProviderCmkClient is that now we can introduce a circular dependency between executables -> cloudstack -> executables. The executables.Cmk interface served to decouple the two packages as I understand so I'm hesitant to break this pattern

Reply:
It seems like `executables.Cmk` is a struct, right?

Either way, in general, we want to define interfaces in the consumer, so in this case that would be in the cloudstack provide package.

What I would recommend in this case is store the executables as `map[string]*executables.Cmk` in the deps struct and then convert the map to ` map[string]cloudstack.ProviderCmkClient` before injecting it in the provider.